### ISSUE TITLE
Modules - No provider "azurerm" plugins meet the constraint "getting started - azure" example does not work with terraform 0.12

### ISSUE THREAD

**Issue by bgarcial** [Azure/terraform-azurerm-compute#104]
- Terraform Version:
```
⟩ terraform -v
Terraform v0.12.1
```

- OS Version:
Linux Mint

## Bug description:

I am using[ the Getting Stated Azure guideline, and I want to create a compute and a network modules](https://learn.hashicorp.com/terraform/azure/modules_az) of this way:

- Azure RM Network Module to create a vnet and subnet.
- Azure RM Compute Module to create a Linux vm.

```hcl
# Use the network module to create a vnet and subnet
module "network" {
    source              = "Azure/network/azurerm"
    version             = "2.0.0"
    location            = "${var.location}"
    resource_group_name = "${azurerm_resource_group.rg.name}"
    address_space       = "10.0.0.0/16"
    subnet_names        = ["mySubnet"]
    subnet_prefixes     = ["10.0.1.0/24"]
}

# Use the compute module to create the VM
module "compute" {
    source            = "Azure/compute/azurerm"
    version           = "1.2.0"
    location          = "${var.location}"
    vnet_subnet_id    = "${element(module.network.vnet_subnets, 0)}"
    admin_username    = "plankton"
    admin_password    = "Password1234!"
    remote_port       = "22"
    vm_os_simple      = "UbuntuServer"
    vm_size           = "${lookup(var.vm_size, var.environment)}"
    public_ip_dns     = ["zzdns"]
}
```

## Debug Output
But when I apply terraform init I get this:
```
⟩ terraform init
Initializing modules...

Initializing the backend...

Initializing provider plugins...
- Checking for available provider plugins...

Provider "random" v1.3.1 is not compatible with Terraform 0.12.1.

Provider version 2.1.0 is the earliest compatible version. Select it with 
the following version constraint:

	version = "~> 2.1"

Terraform checked all of the plugin versions matching the given constraint:
    ~> 1.0

Consult the documentation for this provider for more information on
compatibility between provider and Terraform versions.


No provider "azurerm" plugins meet the constraint "=2.0.0,>= 1.1.0".

The version constraint is derived from the "version" argument within the
provider "azurerm" block in configuration. Child modules may also apply
provider version constraints. To view the provider versions requested by each
module in the current configuration, run "terraform providers".

To proceed, the version constraints for this provider must be relaxed by
either adjusting or removing the "version" argument in the provider blocks
throughout the configuration.


Error: incompatible provider version



Error: no suitable version is available


[I] 

```

## Expected Behavior

I expect that the modules version being downloaded and its works with the terraform 0.12.1 version, allowing initialize the provider plugins and initialize terraform itsef 

### Actual Behavior

Looks like `azurerm` provider is not compatible with 0.12.1 terraform version in order to  download and initialize the plugins providers and modules itself.

Steps to reproduce:

- Please follow [this small part of the guide](https://learn.hashicorp.com/terraform/azure/modules_az) 

I am not using any provider explicitly, but when I add 

```
provider "azurerm" {
  version = "=2.0.0"
}
```
I get that output indicated previously. 

Even if I am using `1.28.0` azure provider version I get the same output.

It is a current error related with the azure provider version or does the modules causing some problem here? 

### References

- [Announcement: Upcoming Changes in Version 2.0 of the Azure Provider](https://github.com/terraform-providers/terraform-provider-azurerm/issues/2807)

- [Plugin installer doesn't filter by protocol version during installation ("Incompatible API version with plugin")](https://github.com/hashicorp/terraform/issues/19221) 

- [Provider incompatible with terraform 0.12.x](https://github.com/terraform-providers/terraform-provider-azurerm/issues/2686)

**Comment by jemag** [comment]
Same problem here. The random provider is already at version 2.1.2. I think it would make a lot of sense to upgrade the dependency of this module to a more recent version of the random provider.

**Comment by lukevenediger** [comment]
Just ran into this problem today. There's a work-around mentioned here https://github.com/hashicorp/terraform/issues/21673#issuecomment-508471742, but it requires you to fork / download this repo and modify the module.

**Comment by GMinchAQ** [comment]
I'm presently updating Getting Started for Terraform 0.12, and just hit this same problem. The fix mentioned in the other thread is correct. You don't really need to fork the repo just to make the change for the Modules lesson. Run terraform init, let it download the modules and error out. Modify main.tf as described in https://github.com/hashicorp/terraform/issues/21673#issuecomment-508471742. Rerun terraform init, and you should be OK. I'll follow up with the module owner to see about getting this fixed.

**Comment by yupwei68** [comment]
Thanks for you replies. This problem has been fixed in [v1.3.0](https://github.com/Azure/terraform-azurerm-compute/releases/tag/v1.3.0). And I'm going to close this issue.
